lldb-eval/context.cc

```diff
@@ -238,6 +238,10 @@ std::unique_ptr<ParserContext::IdentifierInfo> Context::LookupIdentifier(
     } else {
       // In a "value" scope `this` refers to the scope object itself.
       if (name_ref == "this") {
+        // If `scope_` is a reference, dereference it first.
+        if (scope_.GetType().IsReferenceType()) {
+          scope_ = scope_.Dereference();
+        }
```

Comment:
This will modify `scope_` for future calls. If that's okay, should we just do this in the constructor?
Alternatively, don't modify the scope.

Reply:
Oops, good point. Decided to dereference it in constructor.